lib/Model/Importer/Binary/ELFImporter.cpp

```diff
@@ -326,9 +328,76 @@ Error ELFImporter<T, HasAddend>::import(bool FetchDebugInfo) {
   DwarfImporter Importer(Model);
   Importer.import(TheBinary.getFileName(), FetchDebugInfo);
 
+  // Now we try to find missing types in the dependencies.
+  if (FetchDebugInfo > 1)
+    findMissingTypes(TheELF, FetchDebugInfo);
+
   return Error::success();
 }
 
+template<typename T, bool HasAddend>
+void ELFImporter<T, HasAddend>::findMissingTypes(object::ELFFile<T> &TheELF,
+                                                 unsigned FetchDebugInfo) {
+  ModelMap ModelsOfLibraries;
+
+  LDDTree Dependencies;
+  lddtree(Dependencies, TheBinary.getFileName().str(), FetchDebugInfo);
+  for (auto &Library : Dependencies) {
+    revng_log(ELFImporterLog,
+              "Importing Models for dependencies of " << Library.first
+                                                      << ":\n");
+    for (auto &DependencyLibrary : Library.second) {
+      if (ModelsOfLibraries.count(DependencyLibrary))
+        continue;
+      revng_log(ELFImporterLog,
+                " Importing Model for: " << DependencyLibrary << "\n");
+      auto BinaryOrErr = llvm::object::createBinary(DependencyLibrary);
+      if (not BinaryOrErr) {
+        llvm::consumeError(BinaryOrErr.takeError());
+        revng_log(ELFImporterLog,
+                  "Can't create object for " << DependencyLibrary);
+        return;
+      }
+
+      auto &Object = *cast<llvm::object::ObjectFile>(BinaryOrErr->getBinary());
+      auto *TheBinary = dyn_cast<ELFObjectFileBase>(&Object);
+      if (!TheBinary)
+        continue;
+
+      using model::Binary;
+      using std::make_unique;
+      ModelsOfLibraries[DependencyLibrary] = make_unique<TupleTree<Binary>>();
+      TupleTree<model::Binary> &DepModel = *ModelsOfLibraries[DependencyLibrary]
+                                              .get();
+      DepModel->Architecture = Model->Architecture;
+
+      if (auto E = importELF(DepModel,
+                             *TheBinary,
+                             BaseAddress,
+                             1 /*FetchDebugInfo*/)) {
```

Comment:
Do we have recursion here?
Could one craft a binary leading to infinite recursion?

Reply:
Well, I am not sure if it is possible. At least, we are just importing the binary, without looking for missing types (```1 /*FetchDebugInfo*/```).

Reply:
Due to `if (FetchDebugInfoWithLevel > 1)` you mean?

OK, but then `FetchDebugInfoWithLevel == 1` means "don't look in any dependency", right?
If so, make sure this is documented, it's not obvious.

Reply:
Documented in the option description.